src/azure-cli/azure/cli/command_modules/vm/_actions.py

```diff
@@ -45,8 +46,11 @@ def _get_thread_count():
     return 5  # don't increase too much till https://github.com/Azure/msrestazure-for-python/issues/6 is fixed
 
 
-def load_images_thru_services(cli_ctx, publisher, offer, sku, location):
+def load_images_thru_services(cli_ctx, publisher, offer, sku, location, edge_zone):
     from concurrent.futures import ThreadPoolExecutor, as_completed
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.cli.core.profiles import ResourceType
+    edge_zone_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_COMPUTE).virtual_machine_images_edge_zone
```

Comment:
Please put the definition of `edge_zone_client` after the logic of judging `if edge_zone is not None`, otherwise, an error will appear in the old profile because the current api-version does not support this operation

Reply:
changed